driver-legacy/build.gradle

```diff
@@ -30,11 +30,20 @@ test {
 }
 
 dependencies {
-    compile project(':bson')
-    compile project(':driver-core')
-    compile project(':driver-sync')
+    implementation project(path: ':bson', configuration: 'archives')
```

Comment:
Seems like these should all be `api` not `implementation`

Reply:
I will address this in JAVA-3574.